### ISSUE TITLE
disabling 'use change addresses' in confirm_tx_dialog still uses change addresses

### ISSUE THREAD

**Issue by accumulator** [spesmilo/electrum#8769]
1. create tx with a change output
2. observe one output is from change addresses
3. toggle option 'use change addresses' to off
4. observe one output is still from change addresses

**Comment by SomberNight** [comment]
> toggle option 'use change addresses' to off

The change needs to be sent somewhere.
If you turn the setting off, it will reuse one of the "from addresses" and send it there. So if that address happens to be a change address, then the change will be sent to a change address.
Does this match what you observed?

**Comment by accumulator** [comment]
Well, I expected receive/external addresses to be used instead of change/internal addresses..

**Comment by SomberNight** [comment]
Yeah, well, the whole "use_change" setting is a questionable weird thing.

- it needs separate discussion for HD wallets vs imported wallets
- for imported wallets, it is not so clear where change should be sent to
  - hence "use_change" defaults to off, and we just reuse the "from address"
  - if enabled, the change is sent to a random other unused imported addresses. (see https://github.com/spesmilo/electrum/pull/7330)
    - but note that imported addresses/keys are not necessarily homogeneous semantically, and the wallet cannot really distinguish them in general
      - for example, imagine mixing different script types in same wallet
      - or just keys that the user wants to treat differently
      - e.g weird but consider having an online wallet + offline wallet setup, where the online wallet is a superset of the offline wallet, e.g. the user imported also some foreign addresses just for testing into the online one, but of course not into the offline one, and then they create a tx in the online wallet, which then sends the change to an address the offline wallet does not have the key for...
- for deterministic wallets, obviously "use_change" defaults to on
  - if enabled, change is sent to an unused change address from the wallet
  - if disabled, change is sent to one of the ismine "from addresses"
  - this has come up several times over the years, but I cannot find references now, and I usually can't remember the "valid reason" but basically there is no good reason I think to even have the setting at all... it should just be force enabled for HD wallets
    - there are a few loud users who for some reason really don't want to see the coins going to a new address, even after learning about change addresses
      - I can't really find examples now, but I think I saw this on irc, and on forums
    - e.g. in https://github.com/spesmilo/electrum/issues/6463, an RPC user is asking for this option (so likely not a casual user), though without justification

also see https://github.com/spesmilo/electrum/issues/5353
